src/ngraph/function.cpp

```diff
@@ -85,24 +85,28 @@ void Function::init()
 {
     validate_nodes_and_infer_types();
 
-    traverse_nodes(this, [&](shared_ptr<Node> node) {
-        std::shared_ptr<op::Parameter> p = std::dynamic_pointer_cast<op::Parameter>(node);
-        if (nullptr != p)
-        {
-            auto it = std::find_if(m_parameters.begin(),
-                                   m_parameters.end(),
-                                   [p](std::shared_ptr<op::Parameter> q) { return (p == q); });
-            if (it == m_parameters.end())
-            {
-                throw ngraph_error("Function references undeclared parameter");
-            }
-        }
-    });
+    traverse_nodes(this,
+                   [&](shared_ptr<Node> node) {
+                       if (node->is_parameter())
+                       {
+                           auto it = std::find(m_parameters.begin(), m_parameters.end(), node);
+                           if (it == m_parameters.end())
+                           {
+                               throw ngraph_error("Function references undeclared parameter");
+                           }
+                       }
+                   },
+                   true /*include control dependencies*/);
 }
 
```

Comment:
Why not add the control dependencies as an arg to `Function::get_ordered_ops` and pass it along to the sort?